### ISSUE TITLE
N参数问题存疑

### ISSUE THREAD

**Issue by TCcjx** [HuangRiKui/TempValid#5]
在超参数设置中，Hyperparameter Setting. Learning rate lr, batch
size b and number of negative samples N are obtained by grid search based on the MRR of the
validation set. Ultimately, 0.01 lr and 128 b are
applied to all datasets, except ICEWS05-15 which
uses 1024 batch size. For N, 50, 50, 50, 90, 70, and
60 are set for YAGO, WIKI, GDELT ICEWS14,
ICEWS18 and ICEWS05-15, separately.
但是附录中，又说
For rule-adversarial negative samples, we generate feature vectors for the candidate entities reachable by TLogic for training. For ICEWS14, YAGO
and WIKI, we generate the feature vectors of top
100 candidate entities. For ICEWS18, ICEWS05-
15 and GDELT, we generate the feature vectors of
top 50 candidate entities. If the number of negative
samples is still insufficient, we supplement it with
the time-aware negative sampling strategy
但是在训练文件中 neg_num参数并未实际参与训练，请作者解释一下，是我理解有什么地方不对吗？十分期待作者的回复。

**Comment by HuangRiKui** [comment]
> 在超参数设置中，Hyperparameter Setting. Learning rate lr, batch size b and number of negative samples N are obtained by grid search based on the MRR of the validation set. Ultimately, 0.01 lr and 128 b are applied to all datasets, except ICEWS05-15 which uses 1024 batch size. For N, 50, 50, 50, 90, 70, and 60 are set for YAGO, WIKI, GDELT ICEWS14, ICEWS18 and ICEWS05-15, separately. 但是附录中，又说 For rule-adversarial negative samples, we generate feature vectors for the candidate entities reachable by TLogic for training. For ICEWS14, YAGO and WIKI, we generate the feature vectors of top 100 candidate entities. For ICEWS18, ICEWS05- 15 and GDELT, we generate the feature vectors of top 50 candidate entities. If the number of negative samples is still insufficient, we supplement it with the time-aware negative sampling strategy 但是在训练文件中 neg_num参数并未实际参与训练，请作者解释一下，是我理解有什么地方不对吗？十分期待作者的回复。

dataloader中加载了预先计算得到的负样本的向量，在计算损失函数的时候这些负样本参与了优化。
